types/field_type.go

```diff
@@ -105,13 +105,39 @@ func (ft *FieldType) Init(tp byte) {
 	ft.Decimal = UnspecifiedLength
 }
 
+// GetDefaultFieldLengthAndDecimal returns the default display length (flen)
+// and decimal length for column,  taking into account the sign.
+// Refer: source of MySQL 5.7, adjust_max_effective_column_length()
+func (ft *FieldType) GetDefaultFieldLengthAndDecimal() (flen int, decimal int) {
+	flen, decimal = mysql.GetDefaultFieldLengthAndDecimal(ft.Tp)
+	if !mysql.HasUnsignedFlag(ft.Flag) {
+		return
+	}
+
+	switch ft.Tp {
+	case mysql.TypeInt24:
+		fallthrough
+	case mysql.TypeLong:
+		fallthrough
+	case mysql.TypeTiny:
+		fallthrough
+	case mysql.TypeShort:
```

Comment:
You could write `case mysql.TypeInt24, mysql.TypeLong, mysql.TypeTiny, mysql.TypeShort:` instead of using three `fallthrough`s...

Reply:
thx, i adjusted the style